### DISCUSSION TITLE
Most helpful last 30 days

### DISCUSSION THREAD

**Discussion by ghost** [community/community#2887] [Discussions]
The rolling window "Most helpful last 30 days" is probably not so applicable to most repos. It really only makes sense for huge repos with tons and tons of traffic (I guess?). For smaller repos that list is always going to be pretty much empty if not one or two entires. But because anyone can make that list in a small enough repo, it could make sense to support having "Most helpful" (ever) so that you can get an actual overview of whom is helping the most overall.

Maybe you already support this? I can't find this.

**Comment by becca** [comment]
Right now the default is last 30 days and there is not a way to configure/change this. 

If you could change anything about this section what would it be and why? What timeframe would you select, for example, and would it differ per repo? If so, what's the criteria for those repos to get different rolling windows?

**Reply by czyzby** [reply]
I agree that smaller communities have way less traffic for the last 30 days to be useful. I'd consider setting it to 3 months - 1 year in one of my repositories.

**Reply by markseuffert** [reply]
I would occasionally like to switch the timeframe between 30 days and 1 year, in order to see the most helpful contributors at the end of the year. As an alternative, discussions could be shown in a repository's [pulse](https://github.com/github/feedback/pulse/monthly) including the most helpful contributors. Here's an example:

![screenshot-pulse-with-discussions](https://user-images.githubusercontent.com/2429165/151635231-79c0eb56-7023-47e6-909b-81aea1f72878.png)

**Reply by markseuffert** [reply]
See also #10670

**Comment by marcvanzee** [comment]
+1 

I agree 30 days is not very helpful and being able to adjust this would  be very useful.

**Comment by pwalleni** [comment]
@becca your source of inspiration for improving this feature should be this: https://stackoverflow.com/help/whats-reputation

**Comment by icyJoseph** [comment]
Just throwing my two cents here. I noticed this section about a month ago, and since then I've started to see the discussions thread as a game. Whenever someone asks a question, I often try to snipe it and answer, as fast and as detailed as possible. 

I'll probably lose the initial hype rather soon, but I reckon, as a user, I'd like to query the Graphql API later on, and see how many answers I have given to a certain project. Is that possible?

Otherwise, this was a really good addition I believe.

**Edit**: It is possible.

```graphql
  repositoryDiscussionComments (onlyAnswers: true){
      totalCount
    }
```
